connector-schemas/kafka/connection.json

```diff
@@ -49,6 +49,11 @@
                     "additionalProperties": false
                 }
             ]
+        },
+        "groupId": {
```

Comment:
For clustered systems like Kafka, we have two different configurations: the connection profile, and the table. The connection profile provides a single place to configure shared cluster configurations that will be reused across many different instances. So for Kafka, that includes stuff like the bootstrap servers and authentication.

The table configuration is used for specific instances of the cluster. For Kafka this is stuff like the topic name and whether it's a source or sink. 

Since the group id is something you might want to be specific to a topic rather than across the entire cluster, I'd move to the table.json. And since it's only used for sources, it should probably be under the source configuration there.

Reply:
Makes sense, have implemented this.